packages/lib/server/service/insightsBooking.ts

```diff
@@ -1,13 +1,32 @@
 import { Prisma } from "@prisma/client";
+import md5 from "md5";
 import { z } from "zod";
 
-import type { DateRange } from "@calcom/features/insights/server/events";
+import dayjs from "@calcom/dayjs";
+import type { DateRange } from "@calcom/features/insights/server/insightsDateUtils";
 import type { readonlyPrisma } from "@calcom/prisma";
 import { MembershipRole } from "@calcom/prisma/enums";
 
 import { MembershipRepository } from "../repository/membership";
 import { TeamRepository } from "../repository/team";
 
+// Utility function to build user hash map with avatar URL fallback
+export const buildHashMapForUsers = <
```

Comment:
the logics added in this file are almost moved from trpc-router.ts, but the main queries are mostly converted to raw query though